src/meta.rs

```diff
@@ -0,0 +1,274 @@
+use std::collections::HashMap;
+
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct Meta {
```

Comment:
This metadata is path-based (it reminds me of json path selectors), maybe we can somehow include that in the name

Reply:
I'd be as simple as calling it `PathMeta`, is that okay?

Reply:
Yes :)